docs/README.md

```diff
@@ -0,0 +1,110 @@
+## Controllers
+### Clustermesh Controller
+#### Description
+This is the controller responsible for managing the mesh between the Clusters. It watches `clusters.cluster.x-k8s.io` and create the necessary cloud resources to establish the mesh between them leveraging on Crossplane CRs. 
+
+#### How to use
+To add a Cluster to a mesh it needs to have a label `clusterGroup` which is the name of the mesh it will be part of and an annotation `clustermesh.infrastructure.wildlife.io: true` to enable it.
+
+#### Diagram
+![Diagram](./clustermesh/clustermesh-controller.png)
+
+#### Deep Dive
+The controller watches for changes in the Cluster CRs and in order to avoid the complexity of having a broad view of everything in the mesh it only worries about the specific cluster that is being reconcilied at the moment, that means that will only create/update the cloud resources for the current cluster. The trick to make this work without generating drifts or missing resources is that we enqueue every other cluster that belongs to the mesh when a cluster is reconciled. This way we can focus only in the current cluster during the reconciliation.
+
+The cloud resources are created using Crossplane CRs and mapped using Kubernetes ownership relationships to be easier to retrieve and cleanup them. The resources are also mapped in the status of the clustermesh cr for visibility.
+
+The normal reconciliation have four main steps:
+- Retrieve the cluster information
+- Reconcile the VPC peerings
+- Reconcile the Security Groups
+- Reconcile the Routes
+
+##### Retrieve the cluster information
+When a reconciliation is triggered the controller retrieves the necessary information of the current cluster. If the cluster already belongs to the mesh it will update that information, if not it'll only update it. 
+
+##### Reconcile the VPC peerings
+A VPC peering connection is a one to one relationship between two VPCs, because of that we create the peering using <origin-cluster>-<destination-cluster> as the name of the peering so we can easily identify if the peering between two VPCs are already created. 
```

Comment:
I think the text `<origin-cluster>-<destination-cluster>` <origin-cluster> and <destination-cluster> are not appearing in the docs.